annot/src/main/java/com/predic8/membrane/annot/beanregistry/BeanRegistryImplementation.java

```diff
@@ -13,40 +13,44 @@
    limitations under the License. */
 package com.predic8.membrane.annot.beanregistry;
 
-import com.predic8.membrane.annot.Grammar;
-import com.predic8.membrane.annot.bean.BeanFactory;
-import com.predic8.membrane.annot.yaml.GenericYamlParser;
-import com.predic8.membrane.annot.yaml.WatchAction;
+import com.predic8.membrane.annot.*;
+import com.predic8.membrane.annot.bean.*;
+import com.predic8.membrane.annot.yaml.*;
 import org.jetbrains.annotations.*;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.slf4j.*;
 
 import java.util.*;
-import java.util.concurrent.ConcurrentHashMap;
-
+import java.util.concurrent.*;
+import java.util.function.*;
+
+/**
+ * TODO:
+ * - More Tests
+ * - Document
+ * - Do we need uuid when then name is unique?
+ * - For TB Unclear: Lifecycle activation/resolve
+ */
 public class BeanRegistryImplementation implements BeanRegistry, BeanCollector {
 
     private static final Logger log = LoggerFactory.getLogger(BeanRegistryImplementation.class);
 
     private final BeanCacheObserver observer;
     private final Grammar grammar;
 
-    // uid -> bean
-    private final ConcurrentHashMap<String, Object> singletonBeans = new ConcurrentHashMap<>(); // Order is here not critical
```

Comment:
`singletonBeans` kept the *old* instance, e.g. in case of the modification of an `api`. it is required to implement `getOldBean` correctly.